src/main/java/duke/util/DateFormatter.java

```diff
@@ -0,0 +1,66 @@
+package duke.util;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import duke.exception.DukeException;
+
+/**
+ * Contains utility methods for converting between Dates and Strings.
+ */
+public class DateFormatter {
+    // @@author aizatazhar-reused
+    // Reused from https://stackoverflow.com/a/16990333/12003017 with minor modifications
+    private static List<String> dateFormats = Arrays.asList(
+            "dd-MM-yyyy hhmm",
+            "dd-MM-yyyy"
+    );
+
+    /**
+     * Parses a String representation of a date and returns a Date object.
+     *
+     * @param strDate String representation of a date.
+     * @return Date object corresponding to the String given.
+     * @throws DukeException If the String representation does not match any defined date format.
+     */
+    public static Date extractTimestampInput(String strDate) throws DukeException {
+        for (String format: dateFormats) {
+            SimpleDateFormat sdf = new SimpleDateFormat(format);
```

Comment:
Consider a more meaningful name like "formatter"

Reply:
Agree that a more meaningful name here would be better. Thanks